src/components/Identity/createComponent.js

```diff
@@ -150,6 +129,27 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
     commands: {
       setCustomerIds(options) {
         return customerIds.sync(options);
+      },
+      getEcid() {
+        if (ecid) {
+          return ecid;
+        }
+        if (deferredForIdentityCookie) {
+          // Should execute only after the first call resloves
+          return accommodateIdentityOnRequest().then(() => ecid);
+        }
+        if (!hasIdentityCookie()) {
+          // Should defer all calls if an
+          deferredForIdentityCookie = defer();
+        }
+        return getEcidCommand(
```

Comment:
I think this will cause a race condition. If the getEcid request is sent, then a data collection request is sent before the getEcid response is returned from the server, then the user will be identified as two different users (two different ECIDs will have been created). I'm working on a refactor of Identity and I'm trying to take this into account.